beacon-chain/core/blocks/block_operations.go

```diff
@@ -0,0 +1,48 @@
+package blocks
+
+import (
+	"bytes"
+
+	"github.com/sirupsen/logrus"
+
+	v "github.com/prysmaticlabs/prysm/beacon-chain/core/validators"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+var log = logrus.WithField("prefix", "state")
+
+// ProcessProposerSlashings is one of the operations performed
+// on each processed beacon block to penalize proposers based on
+// slashing conditions if any slashable events occurred.
```

Comment:
Please add to this what the spec definition is like 

```
// Spec definition
//   Verify that len(block.body.proposer_slashings) <= MAX_PROPOSER_SLASHINGS.
//   
//   For each proposer_slashing in block.body.proposer_slashings:
//   
//   Let proposer = state.validator_registry[proposer_slashing.proposer_index].
//   Verify that bls_verify(pubkey=proposer.pubkey, msg=hash_tree_root(proposer_slashing.proposal_data_1), sig=proposer_slashing.proposal_signature_1, domain=get_domain(state.fork_data, proposer_slashing.proposal_data_1.slot, DOMAIN_PROPOSAL)).
//   Verify that bls_verify(pubkey=proposer.pubkey, msg=hash_tree_root(proposer_slashing.proposal_data_2), sig=proposer_slashing.proposal_signature_2, domain=get_domain(state.fork_data, proposer_slashing.proposal_data_2.slot, DOMAIN_PROPOSAL)).
//   Verify that proposer_slashing.proposal_data_1.slot == proposer_slashing.proposal_data_2.slot.
//   Verify that proposer_slashing.proposal_data_1.shard == proposer_slashing.proposal_data_2.shard.
//   Verify that proposer_slashing.proposal_data_1.block_root != proposer_slashing.proposal_data_2.block_root.
//   Verify that proposer.status != EXITED_WITH_PENALTY.
//   Run update_validator_status(state, proposer_slashing.proposer_index, new_status=EXITED_WITH_PENALTY).
```

This will put it in a nice code block in godcs like this: https://godoc.org/github.com/fluhus/godoc-tricks#Code_blocks

Additionally gives the reader the spec definition right there (so we can also see if the implementation is outdate rather quickly).

Reply:
Please consider this to make our godocs better.